### PR TITLE
list syscalls supported by the local kernel but not in any syscall groups explicitly in "systemd-analyze syscall-filter"

### PR DESCRIPTION
**Pull request by poettering**
This is really useful to track down missing entries. And it works, as I found three missing entries this way.

### CODE REVIEW

File: src/analyze/analyze.c [hunk 1]
```diff
@@ -1587,8 +1595,9 @@ static int dump_syscall_filters(int argc, char *argv[], void *userdata) {
                         const char *syscall;
                         Iterator j;
 
-                        printf("\nUNLISTED\n"
-                               "    # Supported by local kernel, but not included in any of the groups listed above\n");
+                        printf("\n%sUNLISTED%s\n"
```
Comment (keszybz) [inline src/analyze/analyze.c]: What about turning this into a pseudo-group `@UNLISTED` or so? Right now there's now way to request just the unlisted syscalls.

I also think this should be mentioned in the man page, that if there are sufficient privs, this group will be shown.

If `@UNLISTED` is requested explicitly, and privs are missing, an error should be shown.

Comment (poettering) [reply src/analyze/analyze.c]: So, I was thinking about doing that, but I was afraid that people might think this was a real group they could use in `SystemCallFilter=~@unlisted`, but I am pretty sure we shouldn't do that, as it creates profiles that vary greatly in their effect.

Thinking about this, i think we should probably comment the whole list for this, to make clear that this is not in the same category as the rest of the data.

**Comment by poettering** [comment]
So I force pushed a new version now. it kinda goes the opposite direction that you suggested: in the output the non-listed syscalls are now shown in "commented" form, to make clearer that they are in fact not a group like the others. I hope this communicates clearly that this is not a group you can actually use in SystemCallFilter=, but just some additional commentery. I hope this makes sense?

File: src/analyze/analyze.c [hunk 2]
```diff
@@ -1512,14 +1576,36 @@ static int dump_syscall_filters(int argc, char *argv[], void *userdata) {
...
+                if (k < 0) {
+                        fputc('\n', stdout);
+                        fflush(stdout);
+                        log_notice_errno(k, "# Not showing unlisted system calls, couldn't retrieve kernel system call list: %m");
+                } else if (!set_isempty(kernel)) {
+                        const char *syscall;
+                        Iterator j;
+
+                        printf("\n"
+                               "# %sUnlisted System Calls%s (Spported by the local kernel, but not included in any of the groups listed above)\n",
```
Comment (keszybz) [inline src/analyze/analyze.c]: Typo.

**Comment by poettering** [comment]
Force pushed a new version. Only change is the trivial typo fix @keszybz found. Leaving green label on, since it's so simple.

**Pull request merged**